### PR TITLE
fix(deps): Update and pin Autocomplete to the latest version

### PR DESCRIPTION
**Pull request by shortcuts**
**Summary**

We introduced the required `sourceId` prop in https://github.com/algolia/autocomplete/pull/429 and _I assume that_ as the Autocomplete version wasn't pinned, the latest DocSearch version throw an error.

This implementation provides a default `sourceId` to the DocSearch sources.

_Other improvements might be needed but this should fix the current issue._

### CODE REVIEW

File: packages/docsearch-react/src/DocSearchModal.tsx [hunk 1]
```diff
@@ -139,6 +139,7 @@ export function DocSearchModal({
 
             return [
               {
+                sourceId: 'DocSearchRecentSearches',
```
Comment (francoischalifour) [inline packages/docsearch-react/src/DocSearchModal.tsx:142]: From algolia/autocomplete#490 we'll recommend naming those as `recentSearches` to pick the `sourceId`s easily with destructuring. We can start naming those following this convention already.

File: packages/docsearch-react/src/DocSearchModal.tsx [hunk 2]
```diff
@@ -239,6 +241,7 @@ export function DocSearchModal({
 
               return Object.values<DocSearchHit[]>(sources).map((items) => {
                 return {
+                  sourceId: 'DocSearchHits',
```
Comment (francoischalifour) [inline packages/docsearch-react/src/DocSearchModal.tsx:244]: All the DocSearch section will get the same `sourceId`s here. Can we differentiate them?

**Review by francoischalifour — COMMENTED** [review]
Thanks!

- We should be safe removing the [`enterKeyHint` line](https://github.com/algolia/docsearch/blob/30a7a229ff9490d2fea567b92c1404489f90cc43/packages/docsearch-react/src/SearchBox.tsx#L65) which is [now supported in Autocomplete alpha 38](https://github.com/algolia/autocomplete/blob/next/CHANGELOG.md#features-6).
- We need to update `defaultSelectedItemId` to `defaultActiveItemId`, otherwise the first result is not automatically active. (Why didn't we get a TypeScript error?)

File: packages/docsearch-react/src/DocSearchModal.tsx (hunk 1 above)
Comment (shortcuts) [reply packages/docsearch-react/src/DocSearchModal.tsx:142]: Make sense!

File: packages/docsearch-react/src/DocSearchModal.tsx (hunk 2 above)
Comment (shortcuts) [reply packages/docsearch-react/src/DocSearchModal.tsx]: Fixed in https://github.com/algolia/docsearch/pull/999/commits/fc965daeb4877a3febd619012dff39febc7f5de7

File: packages/docsearch-react/src/DocSearchModal.tsx (hunk 1 above)
Comment (shortcuts) [reply packages/docsearch-react/src/DocSearchModal.tsx]: Fixed in https://github.com/algolia/docsearch/pull/999/commits/fc965daeb4877a3febd619012dff39febc7f5de7

File: packages/docsearch-react/src/DocSearchModal.tsx [hunk 3]
```diff
@@ -237,43 +239,46 @@ export function DocSearchModal({
...
-                                  siblingItem.type === 'lvl1' &&
-                                  siblingItem.hierarchy.lvl1 ===
-                                    item.hierarchy.lvl1
-                              ),
-                          };
-                        })
+              return Object.values<DocSearchHit[]>(sources).map(
+                (items, index) => {
+                  return {
+                    sourceId: `DocSearch-Hits-${index}`,
```
Comment (francoischalifour) [inline packages/docsearch-react/src/DocSearchModal.tsx:245]:
```suggestion
                    sourceId: `hits${index}`,
```

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**